pilot/pkg/xds/monitoring.go

```diff
@@ -167,6 +167,12 @@ var (
 		"Total number of failures to fetch SDS key and certificate.",
 	)
 
+	pilotSDSCertificateErrorsDetail = monitoring.NewSum(
+		"pilot_sds_errors_total",
```

Comment:
Why do we need a second metric? You can add a tag to an existing metric (pilot_sds_certificate_errors_total I mean) and its backwards compatible?

Reply:
I guess that would work too - my concern was that we may have to increment it once ( without tags ) and also once with tags. Which would be strange in other ways.

For this particular case - it may not matter too much, but in general it's a concern - metric names and labels are 
a de-facto API.

I can change - do you want to increment 2x ( so the unlabeled metric remains consistent ) or just once ?

Reply:
I don't think we need the unlabeled one, any query today would be like `sum(pilot_sds_errors_total{})` which will behave the same with labels added I think?

Actually incrementing twice would break it

Reply:
Ok, will update it. 

Should we add the namespace of the gateway too ( so namespace admins can create alerts/graphs for their own ns) ?

Reply:
Maybe? it has cardinality risks though so probably a bit risk.

Namespace users should be looking at Gatway status, though. If we aren't returning useful errors to that then we should

Reply:
Can we replace \n with something and shove it into status ? It's another option I guess.

That's why I'm looking more into traces - at least frameworks exist to sample, merge or analyze traces, the current
logs are a huge pain and require access to istiod logs for people who shouldn't.

Reply:
However - the key is that metrics allow alerts - Gateway status doesn't. 
If you want oncall to be paged if there are any errors - including secrets distributed to gateways - we need metrics.

( the comment for shoving into status was for the other thread with the bad yaml, got mixed up)